### ISSUE TITLE
Window too tall for a 1024x768 desktop and other height problems

### ISSUE THREAD

**Issue by eltomito** [jliljebl/flowblade#122]
The main window is taller than my desktop height and it resists any attempts to make it shorter. This makes Flowblade really annoying to use on a 1024x768 desktop.

I can't even resize the inside of the window. It looks like there's a drag bar that promises to let me make the tracks taller and the player window shorter, but when I grab it, it doesn't move.
Also, I tried to make a video track "Large Height" and it complained that there isn't enough vertical height. Well, there would be tons of it if I could make the player window use less of it and give more of it to the tracks and there would be even more height available if the track section could be in a vertically scrolled window of it's own.

TL;DR
3 height problems:
1) The whole main window is too tall and cannot be made shorter
2) The track section isn't tall enough and cannot be made taller
3) The track section doesn't scroll vertically when there isn't enough height.

**Comment by denjello** [comment]
I have noticed this too on my laptop with 1366x768 resolution.

**Comment by jliljebl** [comment]
I haven't looked at situation with 768 height for a while. I made it work a few years ago, but maybe new GUI code has broken it.

I'll improve the situation for this. I'm not sure however that I can make 1024 width work correctly

**Comment by jliljebl** [comment]
Ok, this is now fixed in repository in so that 1152 width and 768 height work at least here. 

If there are still problems, please post a comment here with a _SCREENSHOT_ and information on which OS and DE and/or WM is used. I will then re-open issue if needed.

**Comment by faridosc** [comment]
bug continues on my laptop (14" resolution 1366x768). 
archlinux + gnome3.16. 

![captura de tela de 2015-08-03 15-38-59](https://cloud.githubusercontent.com/assets/373726/9044913/34bb37d0-39f6-11e5-9ca9-ec1197df150e.png)

**Comment by jliljebl** [comment]
![screenshot from 2015-08-03 21 59 55](https://cloud.githubusercontent.com/assets/1823444/9045362/47f6028c-3a2b-11e5-90f2-35c4e9bc2df1.png)

Ok, I added patch that force removes Audio meter on the right, I only made it impossible to add it but you had it already active.. The "Render" panel should be rearranged like in the picture. Could you pull from repository again and try again?

**Comment by faridosc** [comment]
much better, but feels a bit weird still. i feel some pixels are eaten at the bottom. also the maximize-minimize button disappears.

![captura de tela de 2015-08-03 17-02-04](https://cloud.githubusercontent.com/assets/373726/9046560/c7cc58fa-3a01-11e5-85cf-aad9362e9de5.png)

**Comment by faridosc** [comment]
i noticed that this happens in gnome... i tried it with pekwm and it works fine.

**Comment by jliljebl** [comment]
Ok, I'll leave it as is for now, and maybe install in VirtualBox Gnome later and see what gives. BTW, seems dark theme colors/icons/buttons could be improved still, I'll give it a go before next release.

**Comment by faridosc** [comment]
i tried it with mate and works fine... a gnome issue probably.

**Comment by techwiz24** [comment]
I can confirm the Maximize button is missing in gnome (which is standard by default unless you re-enable it globally via gnome-tweak). If this is gtk+2, it might be a different gconf setting that needs tweaking, but I am not sure.

**Comment by jliljebl** [comment]
@techwiz24 are you running Gnome 3.16 on system that has screen height of 768px or possibly screen width is 1151px?

**Comment by techwiz24** [comment]
@jliljebl Technically, i think? My primary display is my 1920x1080@60 on my laptop. I have a secondary display connected via HDMI that is 1366x768@60.

**Comment by jliljebl** [comment]
Is maximize missing when application is opened in 1920x1080 screen? Does disconnecting the 1366x768 screen change the situation?

**Comment by techwiz24** [comment]
Ok, that's really odd. If I ***move*** the window to the 1920x1080 display, the maximize button shows up. If it's moved to the 1366x768 display, the maximize button goes away.

@jliljebl To answer your question directly, yes, if only the 1920x1080 display is active, then the maximize button is present.

**Comment by jliljebl** [comment]
Ok, this can be explained. Flowblade is still a few pixels too tall for Gnome on 768px height because Gnome uses a lot of vertical space for tall header bar + screen top stripe.

When you start the application in terminal do get on in the print out  "Small height: False" or "Small height: True"? I get:
---
Use OS locale language.
Small height: False
Small width: False
Detecting environment...
---
What happens if only the  1366x768 is active? Flowblade does optimizations for small heights and if it detects  1920x1080 display then those are omitted.

**Comment by techwiz24** [comment]
When both displays are active, both `Small height` and `Small width` are `false`. When the primary display is not active, `Small height` is true and `Small width` is `false` and the maximize button is not displayed.

I'm curious, is there ever a situation where a maximize button should **not** be displayed?

**Comment by jliljebl** [comment]
> I'm curious, is there ever a situation where a maximize button should not be displayed?

No, Gnome disables maximize when application is too tall for screen.

The solution here is that I would need to pull a few height pixels for Gnome on 768px height screen from...well, I don't know where I'm going to get those. I'll give it a shot for next release, but I'm not completely sure it is easily doable.

**Comment by techwiz24** [comment]
> No, Gnome disables maximize when application is too tall for screen.

Is that a gnome thing or a GTK thing? I can't replicate that with non-GTK applications, but I can replicate with this small sample in GTK+3:

```python
from gi.repository import Gtk
import sys


class App(Gtk.Window):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)

        self.set_titlebar(hb)

        self.set_title("Test Window")
        self.add(Gtk.Button("Test Button"))

        self.set_size_request(800,1200)


if __name__ == '__main__':
    Gtk.init(sys.argv)

    a = App()
    a.show_all()

    Gtk.main()
```

I think this may be an artifact of moving to CSD's. This doesn't happen if I switch to a `HeaderBar` by adding:

```python
hb = Gtk.HeaderBar()
hb.set_show_close_button(True)
```

before 

```python
self.set_title("Test Window")
```

I'm not sure if moving to a `HeaderBar` is something you would consider or not.

Thanks for the quick response by the way!

**Comment by jliljebl** [comment]
Seems a GTK3 thing. Moving to HeaderBar is not on cards because of the firmly trenched design direction of the application. Thanks for comments and feedback on this, at least the issue is now understood.

**Comment by techwiz24** [comment]
As a follow-up now that I got it [running from master](https://github.com/jliljebl/flowblade/issues/157), changing to a headerbar just for the looks (ie: The title and CSD buttons, not menus or any other items on it) does indeed show the maximize button on my secondary display, but it's non-functional:

![screenshot from 2015-09-15 13-56-45](https://cloud.githubusercontent.com/assets/794251/9885189/dd538316-5bb1-11e5-9da5-cfbd71518a8b.png)

I tried in the example app I made the other day and GTK+ had a fun time trying to render that...:

![screenshot from 2015-09-15 13-58-15_cropped](https://cloud.githubusercontent.com/assets/794251/9885250/38531d30-5bb2-11e5-8690-847969dff8c3.png)

I'm going to see if I can talk to the GTK+ folks and possibly file a bug report if one has not been filed already.

**Comment by jliljebl** [comment]
I think HeaderBar (and the CSD it requires) only works nicely on Gnome, and seems that you are having problems with it on Gnome too. I had some old code and tried HeaderBar on my Ubuntu today, and the window controls were some ugly "almost same" looking things, plus the window border was wrong size making the whole thing look decidedly "non-native". This is not working for me.

If same code gives different results in two different languages, then there very well might be bindings bugs.

**Comment by faridosc** [comment]
just a suggestion, could you maybe add an option to toggle flowblade fullscreen in the view menu, maybe that way at least it can occupy the whole screen look/feel better.

**Comment by jliljebl** [comment]
Yeah, this should be easy to do, and it adds immediately value to some users. I'll add this to next release.

**Comment by jliljebl** [comment]
Full screen support is in repository and will be in 1.4

**Comment by faridosc** [comment]
just tried it, feels much better :)

**Comment by qubodup** [comment]
1310x891 is the minimum possible resolution on my system. `wmctrl` can force lower resolutions but that will cut off part of the UI and it will "jump" back to minimum resolution when trying to resize the window.

One reason to support low resolutions: tutorials recorded at low resolutions are more useful to people with only one screen on which they want to both watch and copy actions, as well as more useful to those with small screens. I recommend to support 1280x720

https://help.ubuntu.com/community/Installation/SystemRequirements recommends 1024x768.

1024x720 would be a nice minimum resolution.

**Comment by jliljebl** [comment]
We declare victory. We will create new Issue once we get another one of these.
